core/src/main/java/org/testcontainers/images/AbstractImagePullPolicy.java

```diff
@@ -0,0 +1,42 @@
+package org.testcontainers.images;
+
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.DockerLoggerFactory;
+
+@Slf4j
+public abstract class AbstractImagePullPolicy implements ImagePullPolicy {
+
+    private static final LocalImagesCache LOCAL_IMAGES_CACHE = LocalImagesCache.INSTANCE;
+
+    @Override
+    public boolean shouldPull(DockerImageName imageName) {
+        Logger logger = DockerLoggerFactory.getLogger(imageName.toString());
+
+        // Does our cache already know the image?
+        ImageData imageData = LOCAL_IMAGES_CACHE.get(imageName);
```

Comment:
I'm half tempted to suggest having an additional boolean that can be used in line 27, e.g. something like:
```
if (isCached && !shouldPullCached(...
```

Alternatively, perhaps renaming this to `cachedImageData` would be equivalently clear.